measures/UpgradeCosts/measure.rb

```diff
@@ -0,0 +1,282 @@
+      next unless (not option_lifetimes[option_num].nil?) && (option_lifetimes[option_num] != 0)
+      lifetime = option_lifetimes[option_num].round(2)
+      option_lifetime_name = 'option_%02d_lifetime_yrs' % option_num
+      register_value(runner, option_lifetime_name, lifetime)
+      runner.registerInfo("Registering #{lifetime} for #{option_lifetime_name}.")
+    end
+    upgrade_cost = upgrade_cost.round(2)
+    register_value(runner, upgrade_cost_name, upgrade_cost)
+    runner.registerInfo("Registering #{upgrade_cost} for #{upgrade_cost_name}.")
+
+    return true
+  end
+
+  def get_cost_multiplier(cost_mult_type, hpxml, runner)
+    cost_mult = 0.0
+    if cost_mult_type == 'Fixed (1)'
+      cost_mult += 1.0
+    elsif cost_mult_type == 'Wall Area, Above-Grade, Conditioned (ft^2)'
+      hpxml.walls.each do |wall|
+        next unless wall.is_thermal_boundary
+
+        cost_mult += wall.area
+      end
+    elsif cost_mult_type == 'Wall Area, Above-Grade, Exterior (ft^2)'
+      hpxml.walls.each do |wall|
+        next unless wall.is_exterior
+
+        cost_mult += wall.area
+      end
+    elsif cost_mult_type == 'Wall Area, Below-Grade (ft^2)'
+      hpxml.foundation_walls.each do |foundation_wall|
+        cost_mult += foundation_wall.area
+      end
+    elsif cost_mult_type == 'Floor Area, Conditioned (ft^2)'
+      cost_mult += hpxml.building_construction.conditioned_floor_area
+    elsif cost_mult_type == 'Floor Area, Attic (ft^2)'
+      hpxml.frame_floors.each do |frame_floor|
+        next unless frame_floor.is_thermal_boundary
+        next unless frame_floor.is_interior
+
+        if frame_floor.is_floor
+          next unless [HPXML::LocationAtticVented, HPXML::LocationAtticUnvented].include?(frame_floor.interior_adjacent_to)
+        elsif frame_floor.is_ceiling
+          next unless [HPXML::LocationAtticVented, HPXML::LocationAtticUnvented].include?(frame_floor.exterior_adjacent_to)
+        end
+
+        cost_mult += frame_floor.area
+      end
+    elsif cost_mult_type == 'Floor Area, Lighting (ft^2)'
+      cost_mult += hpxml.building_construction.conditioned_floor_area
+      hpxml.slabs.each do |slab|
+        next unless [HPXML::LocationGarage].include?(slab.interior_adjacent_to)
+
+        cost_mult += slab.area
+      end
+    elsif cost_mult_type == 'Roof Area (ft^2)'
+      hpxml.roofs.each do |roof|
+        cost_mult += roof.area
+      end
+    elsif cost_mult_type == 'Window Area (ft^2)'
+      hpxml.windows.each do |window|
+        cost_mult += window.area
+      end
+    elsif cost_mult_type == 'Door Area (ft^2)'
+      hpxml.doors.each do |door|
+        cost_mult += door.area
+      end
+    elsif cost_mult_type == 'Duct Surface Area (ft^2)'
+      hpxml.hvac_distributions.each do |hvac_distribution|
+        hvac_distribution.ducts.each do |duct|
+          cost_mult += duct.duct_surface_area
+        end
+      end
+    elsif cost_mult_type == 'Size, Heating System (kBtu/h)'
```

Comment:
Wouldn't this code end up including the heating supplemental system capacity? I don't see where you are accounting for that.

Reply:
Good point. I added `next if heating_system.fraction_heat_load_served != 1.0`.

Reply:
You could technically have a single heating system/heat pump that serves less than 100%, but maybe ResStock never generates such an HPXML.

Reply:
We may want that ability in the future for zonal electric baseboard.

Reply:
I changed this to check for number of heating systems. If there are exactly 1, then it is the primary heating system (and there is no secondary heating system). If there are exactly 2, then the heating system with the least load fraction served is the secondary heating system.